openapi.yaml

```diff
@@ -1,279 +1,662 @@
+              schema:
+                $ref: "#/components/schemas/LocationsResultObjects"
+        "404":
+          description: "Not Found"
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/Error"
+        "500":
+          description: "Internal Server Error"
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/Errors"
+  /locations/{locationID}:
+    get:
+      tags:
+        - locations
+      description: "Finds location by ID"
+      operationId: getLocationByID
+      parameters:
+        - name: locationID
+          in: path
+          description: "ID of location to be searched by"
+          required: true
           schema:
-            $ref: '#/definitions/PetsResult'
-        '400':
-          $ref: '#/responses/400'
-        '500':
-          $ref: '#/responses/500'
-    post:
-      summary: Add a new pet
-      operationId: postPet
+            type: string
+      responses:
+        "200":
+          description: "Successful response"
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/LocationsResultObject"
+        "500":
+          description: "Internal Server Error"
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/Errors"
+  /locations/{locationID}/services:
+    get:
       tags:
-        - pets
+        - locations
+        - services
+      description: "Lists services related to a location"
+      operationId: getServices
       parameters:
-        - in: body
-          name: body
+        - name: locationID
+          in: path
+          description: "ID of location"
           required: true
           schema:
-            $ref: '#/definitions/PetPostBody'
+            type: string
       responses:
-        '201':
-          description: Pet successfully created
+        "200":
+          description: "Successful response"
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/ServicesResultObjects"
+        "404":
+          description: "Resource not found"
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/Errors"
+        "500":
+          description: "Internal Server Error"
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/Errors"
+  /services:
+    get:
+      tags:
+       - services
+      summary: Get services
+      description: "Gets all `Service` objects. Can be filtered by parameters."
+      operationId: getServicesList
+      parameters:
+        - in: query
+          name: filter[query]
+          description: "Query to search through buildings. Looks through the building name, abbreviation."
```

Comment:
I don't think this description is right (ie it's searching services, not buildings) but I'm not sure what the correct distinction is